libcloud/common/base.py

```diff
@@ -738,8 +748,15 @@ def request(self, action, params=None, data=None, headers=None,
 
                 self.connection.endheaders()
             else:
-                self.connection.request(method=method, url=url, body=data,
-                                        headers=headers)
+                if retry_enabled:
+                    retry(self.connection.request, self.timeout,
```

Comment:
The argument order might be wrong, or you could just make everything but the decorated method be a kwarg instead, which is probably easier to read anyway.